tactics/class_tactics.ml

```diff
@@ -39,10 +37,6 @@ module NamedDecl = Context.Named.Declaration
 let typeclasses_debug = ref 0
 let typeclasses_depth = ref None
 
-let typeclasses_modulo_eta = ref false
```

Comment:
Hmm, I'm a bit worried here, shouldn't we have made it true before?

Reply:
Hum, if we are removing the deprecated option, we have to keep the default value. I followed the doc which said

> This option allows eta-conversion for functions and records during unification of type-classes. This option is unsupported since 8.6 with Typeclasses Filtered Unification set, but still affects the default unification strategy, and the one used in Legacy Resolution mode. It is unset by default. If Typeclasses Filtered Unification is set, this has no effect and unification will find solutions up-to eta conversion. Note however that syntactic pattern-matching is not up-to eta.

Reply:
Right, thanks, makes sense. It is now always up-to eta for conversion.